crates/environ/src/func_environ.rs

```diff
@@ -1180,13 +1103,24 @@ impl<'module_environment> cranelift_wasm::FuncEnvironment for FuncEnvironment<'m
 
     fn translate_table_size(
         &mut self,
-        _pos: FuncCursor,
-        _index: TableIndex,
+        mut pos: FuncCursor,
+        table_index: TableIndex,
         _table: ir::Table,
     ) -> WasmResult<ir::Value> {
-        Err(WasmError::Unsupported(
-            "reference types: `table.size`".to_string(),
-        ))
+        let (func_sig, func_idx, table_index_arg) =
```

Comment:
Presumably we're going to want to have the table size available inline for `table.get`, and in that case we probably won't have an intrinsic, right? 

If that's the case could this go ahead and implement the JIT support necessary to do the load inline?

Reply:
Sure thing.